### PR TITLE
Allow specifying a single configuration

### LINKED ISSUE #790: OTLP Exporter must support single configuration for all signals
A follow-up issue to discussions in https://github.com/open-telemetry/opentelemetry-specification/pull/699

The OTLP exporter must allow the same configuration for metrics/traces (and in the future logs) to support the most common use-case of sending all telemetry to the same collector and multiplexing the data from the collector.

### PR DESCRIPTION
**Pull request by codeboten**
The following change describes that the OTLP exporter must support configuration for all signals via a single set of configuration options. There is also an example for configuring different signals with different endpoints via environment variables.

Fixes #790 

Related issues #699 #681

### CODE REVIEW

File: specification/trace/sdk_exporters/otlp.md [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
+Traces and metrics are sent to different collectors using different protocols:
+
+```bash
+export OTEL_EXPORTER_OTLP_SPAN_ENDPOINT=collector:55680
+export OTEL_EXPORTER_OTLP_SPAN_PROTOCOL=grpc
+export OTEL_EXPORTER_OTLP_SPAN_INSECURE=true
+
+export OTEL_EXPORTER_OTLP_METRICS_ENDPOINT=collector.example.com/v1/metrics
+export OTEL_EXPORTER_OTLP_METRICS_PROTOCOL=http
+```
```
Comment (tigrannajaryan) [inline specification/trace/sdk_exporters/otlp.md:39]: Can we also show a third example which demonstrates the override and what happens when we override one signal but not the other?

File: specification/trace/sdk_exporters/otlp.md [hunk 2]
```diff
@@ -1,20 +1,42 @@
 # OpenTelemetry Protocol Exporter
 
-This document specifies the configuration options available to the OpenTelemetry Protocol ([OTLP](https://github.com/open-telemetry/oteps/blob/master/text/0035-opentelemetry-protocol.md)) `SpanExporter` and `MetricsExporter` as well as the retry behavior.
```
Comment (SergeyKanzhelev) [inline specification/trace/sdk_exporters/otlp.md:3]: should this be moved out of `trace` folder then?

File: specification/trace/sdk_exporters/otlp.md (hunk 1 above)
Comment (codeboten) [reply specification/trace/sdk_exporters/otlp.md:39]: Added a third example

File: specification/trace/sdk_exporters/otlp.md (hunk 2 above)
Comment (codeboten) [reply specification/trace/sdk_exporters/otlp.md:3]: i agree it makes sense to move it out of the trace folder. any thoughts on the best location for this? i can imagine it under `common` or under `protocol`. Although maybe a new `exporters` folder would make more sense.

Comment (SergeyKanzhelev) [reply specification/trace/sdk_exporters/otlp.md:3]: https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/protocol seems to be a good fit from the content point of view.

Perhaps the entire protocol folder may be renamed later into `OTLP` or `exporter/OTLP`.

Comment (codeboten) [reply specification/trace/sdk_exporters/otlp.md:3]: sounds good, moved to protocol directory.

**Comment by SergeyKanzhelev** [comment]
Last small nit: please add a reference to this page from one of high level docs. So it will be discoverable

**Comment by codeboten** [comment]
> I'd really love if the reference to this doc will be added to one of the higher-level docs/TOCs so the page will be discoverable

Link has been added @SergeyKanzhelev

File: specification/protocol/exporter.md [hunk 3]
```diff
@@ -1,20 +1,54 @@
...
+| Insecure             | Whether to enable client transport security for the exporter's `grpc` or `http` connection. | `false`           | `OTEL_EXPORTER_OTLP_INSECURE` `OTEL_EXPORTER_OTLP_SPAN_INSECURE` `OTEL_EXPORTER_OTLP_METRIC_INSECURE` |
+| Certificate File     | Certificate file for TLS credentials of gRPC client. Should only be used if `insecure` is set to `false`. | n/a               | `OTEL_EXPORTER_OTLP_CERTIFICATE` `OTEL_EXPORTER_OTLP_SPAN_CERTIFICATE` `OTEL_EXPORTER_OTLP_METRIC_CERTIFICATE` |
+| Headers              | The headers associated with gRPC or HTTP requests.                   | n/a               | `OTEL_EXPORTER_OTLP_HEADERS` `OTEL_EXPORTER_OTLP_SPAN_HEADERS` `OTEL_EXPORTER_OTLP_METRIC_HEADERS` |
+| Compression          | Compression key for supported compression types. Supported compression: `gzip`| no compression              | `OTEL_EXPORTER_OTLP_COMPRESSION` `OTEL_EXPORTER_OTLP_SPAN_COMPRESSION` `OTEL_EXPORTER_OTLP_METRIC_COMPRESSION` |
+| Timeout              | Max waiting time for the backend to process each spans or metrics batch. | 60s               | `OTEL_EXPORTER_OTLP_TIMEOUT` `OTEL_EXPORTER_OTLP_SPAN_TIMEOUT` `OTEL_EXPORTER_OTLP_METRIC_TIMEOUT` |
```
Comment (bogdandrutu) [inline specification/protocol/exporter.md:17]: As an example I would use `10s` or even `5s`

File: specification/protocol/README.md [hunk 4]
```diff
@@ -5,3 +5,4 @@ This is the specification of new OpenTelemetry protocol (OTLP).
 - [Design Goals](design-goals.md).
 - [Requirements](requirements.md).
 - [Specification](otlp.md).
+- [Exporter](exporter.md).
```
Comment (bogdandrutu) [inline specification/protocol/README.md:8]: Sdk Exporter? Just to clarify that this refers to the SDK exporter, we do have exporters in the collector as well, so just to make sure no confusion.

**Review by carlosalberto — APPROVED** [review]

File: specification/protocol/exporter.md (hunk 3 above)
Comment (codeboten) [reply specification/protocol/exporter.md]: Updated.

File: specification/protocol/README.md (hunk 4 above)
Comment (codeboten) [reply specification/protocol/README.md]: Yes that's correct, I've updated the link to say `SDK Exporter`

**Comment by bogdandrutu** [comment]
non controversial, merging

**Pull request merged**